yt/frontends/ytdata/data_structures.py

```diff
@@ -86,7 +86,7 @@ def _parse_parameter_file(self):
             for key in f.attrs.keys():
                 v = parse_h5_attr(f, key)
                 if key == "con_args":
-                    v = v.astype("str")
+                    v = eval(v)
```

Comment:
This is a backward incompatible change. Old datasets won't be loadable. Is there a reason for this?

Reply:
I did this to fix a test. The ytdata frontend is using the json representation of the unit registry, which I had to change in unyt so I don't think there's a clear way to load old registries written out by yt.units and load them with unyt again.

Are we really supporting loading ytdata datasets accross yt versions? There's no tests for that, for example...